### PR TITLE
setting inodeLimit to 1M for consistencyGroup

### LINKED ISSUE #591: Need large inodelimit for CG independent fileset
## Describe the bug
Need large inodelimit for CG independent fileset 

## How to Reproduce?

Need large inodelimit for CG independent fileset, today it is set to 1Lac in case of <10GB first PVC volume in the namespace(CG)

## Expected behavior
Need large inodelimit for CG independent fileset

### LINKED ISSUE #596: PVC creation failed if compression=false
## Describe the bug
PVC creation failed if compression=false 

## How to Reproduce?
1. Created storageclass with compression=false 

```
[root@csiscale-master ~]# oc describe sc ibm-spectrum-scale-csi-advance-tier-false
Name:            ibm-spectrum-scale-csi-advance-tier-false
IsDefaultClass:  No
Annotations:     kubectl.kubernetes.io/last-applied-configuration={"allowVolumeExpansion":true,"apiVersion":"storage.k8s.io/v1","kind":"StorageClass","metadata":{"annotations":{},"name":"ibm-spectrum-scale-csi-advance-tier-false"},"parameters":{"compression":"false","tier":"system","type":"advanced","volBackendFs":"gpfs0"},"provisioner":"spectrumscale.csi.ibm.com","reclaimPolicy":"Delete"}

Provisioner:           spectrumscale.csi.ibm.com
Parameters:            compression=false,tier=system,type=advanced,volBackendFs=gpfs0
AllowVolumeExpansion:  True
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     Immediate
Events:                <none>

```
2. Created PVC using above storageclass 

```
[root@csiscale-master ~]# oc get pvc
NAME                  STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                                 AGE
scale-pv-tier-false   Pending                                                                        ibm-spectrum-scale-csi-advance-tier-false    8m33s
```
```
[root@csiscale-master ~]# oc describe pvc scale-pv-tier-false
Name:          scale-pv-tier-false
Namespace:     ibm-spectrum-scale-csi-driver
StorageClass:  ibm-spectrum-scale-csi-advance-tier-false
Status:        Pending
Volume:
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: spectrumscale.csi.ibm.com
               volume.kubernetes.io/storage-provisioner: spectrumscale.csi.ibm.com
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:
Access Modes:
VolumeMode:    Filesystem
Used By:       <none>
Events:
  Type     Reason                Age                    From                                                                                                 Message
  ----     ------                ----                   ----                                                                                                 -------
  Warning  ProvisioningFailed    8m41s (x9 over 10m)    persistentvolume-controller                                                                          storageclass.storage.k8s.io "ibm-spectrum-scale-csi-advance-tier-false" not found
  Normal   Provisioning          2m3s (x10 over 8m26s)  spectrumscale.csi.ibm.com_ibm-spectrum-scale-csi-provisioner-0_62a350eb-7d52-4891-9ad7-3dbe3d085dff  External provisioner is provisioning volume for claim "ibm-spectrum-scale-csi-driver/scale-pv-tier-false"
  Warning  ProvisioningFailed    2m3s (x10 over 8m26s)  spectrumscale.csi.ibm.com_ibm-spectrum-scale-csi-provisioner-0_62a350eb-7d52-4891-9ad7-3dbe3d085dff  failed to provision volume with StorageClass "ibm-spectrum-scale-csi-advance-tier-false": rpc error: code = InvalidArgument desc = invalid compression algorithm specified: false
  Normal   ExternalProvisioning  26s (x33 over 8m26s)   persistentvolume-controller                                                                          waiting for a volume to be created, either by external provisioner "spectrumscale.csi.ibm.com" or manually created by system administrator
```
Note : PVC is in pending state 

Logs from driver :

```
I0202 10:36:40.436218       1 gpfs.go:181] gpfs ValidateControllerServiceRequest
I0202 10:36:40.436250       1 gpfs_util.go:346] gpfs_util invalid compression algorithm specified: false
E0202 10:36:40.436264       1 utils.go:55] GRPC error: rpc error: code = InvalidArgument desc = invalid compression algorithm specified: false
I0202 10:36:47.920629       1 utils.go:51] GRPC call: /csi.v1.Identity/Probe
```

## Expected behavior
PVC should be in bound state

### PR DESCRIPTION
**Pull request by deeghuge**
## Pull request checklist

setting inodeLimit to 1M for consistencyGroup is user has not specified the inodeLimit in storageClass
adding validation for compression=false

```release-notes

```
fixes #596 
fixes #591 
## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Review by amdabhad — APPROVED** [review]
LGTM

**Review by Jainbrt — DISMISSED** [review]
LGTM

File: driver/csiplugin/gpfs_util.go [hunk 1]
```diff
@@ -269,12 +269,12 @@ func getScaleVolumeOptions(volOptions map[string]string) (*scaleVolume, error) {
...
 		return &scaleVolume{}, status.Error(codes.InvalidArgument, "The parameters \"type\" and \"volDirBasePath\" are mutually exclusive")
 	}
- 
+
 	if fsTypeSpecified || isSCTypeSpecified {
 		scaleVol.IsFilesetBased = true
 	}
 
-	if isCompressionSpecified && compression == "" {
+	if isCompressionSpecified && (compression == "" || compression == "false") {
```
[the remark this answers is not in the document]
Comment (deeghuge) [reply driver/csiplugin/gpfs_util.go:277]: yes, we do have validation for allowed values later

File: driver/csiplugin/controllerserver.go [hunk 2]
```diff
@@ -322,6 +322,12 @@ func (cs *ScaleControllerServer) createFilesetBasedVol(scVol *scaleVolume, isNew
 		//Set uid and gid as 0 for CG independent fileset
 		opt[connectors.UserSpecifiedUid] = "0"
 		opt[connectors.UserSpecifiedGid] = "0"
+		if scVol.InodeLimit != "" {
+			opt[connectors.UserSpecifiedInodeLimit] = scVol.InodeLimit
+		} else {
+			opt[connectors.UserSpecifiedInodeLimit] = "1M"
+			// Assumption: On an avertage a consistancy group contains 10 volumes
```
[the remark this answers is not in the document]
Comment (deeghuge) [reply driver/csiplugin/controllerserver.go]: done

**Review by madhuthorat — APPROVED** [review]

**Pull request merged**